lib/action_subscriber/railtie.rb

```diff
@@ -3,10 +3,11 @@ class Railtie < ::Rails::Railtie
     config.action_subscriber = ::ActionSubscriber.config
 
     ::ActiveSupport.on_load(:active_record) do
+      require "action_subscriber/middleware/active_record/connection_management_async"
       require "action_subscriber/middleware/active_record/connection_management"
       require "action_subscriber/middleware/active_record/query_cache"
 
-      ::ActionSubscriber.config.middleware.use ::ActionSubscriber::Middleware::ActiveRecord::ConnectionManagement
+      ::ActionSubscriber.config.middleware.use ::ActionSubscriber::Middleware::ActiveRecord::ConnectionManagementAsync
```

Comment:
will changing this default cause problems for downstream dependencies? I'm thinking of despacho which will add its middleware before the `ConnectionManagement` middleware.

That should be pretty easy to fix, but it seems like this makes it a major version bump as opposed to a minor version bump.